designs/2022-forbid-disable/README.md

```diff
@@ -0,0 +1,120 @@
+- Repo: eslint/eslint
+- Start Date: 2022-02-28
+- RFC PR:
+- Authors: Brian Bartels
+
+# ESLintRC Option To Forbid Disable For Specific Rules
+
+## Summary
+
+Allow users to forbid users from disabling certain rules. This is similar to `noInlineConfig` where the primary difference is that `noInlineConfig` is set for all rules.
+
+## Motivation
+
+When working with complex lint configs. Some rules are more important than other. For this reason it can be inconsequential to disable certain rules. But incredibly bad to disable others.
+
+EG:
+```
+//eslint-disable-next-line @government/required-SECURITY-rule
+```
+Is a much worse rule to disable than
+```
+//eslint-disable-next-line noExtraSemi
+```
+
+`noInlineConfig` works well for some. But in reality, many users disable rules at least somewhat frequently for valid reasons. And in large repos, there are going to be exceptions that need inline configuration. 
+
+## Detailed Design
+
+in `linter.js` I will adjust the `getDirectiveComments` function so that I reference a list of forbidden rules. If someone is disabling a rule that is forbidden. I will add an error similar to `warnInlineConfig`
+
+In `eslintrc` I will add another option for rules. The new state can be
+```
+"warn"
+"error"
+"forbid"
```

Comment:
Since this is a new severity value in the configuration, how does it map to [`LintMessage#severity`](https://eslint.org/docs/developer-guide/nodejs-api#-lintmessage-type)? Does it produce lint messages with `severity: 2`, same as `"error"`?

Reply:
it will produce lintMessages with severity: 2